Sources/Noora/Components/Completion.swift

```diff
@@ -0,0 +1,108 @@
+        case let .error(error): return error
+        default: fatalError("Trying to obtain a message from a compound item")
+        }
+    }
+
+    var flattened: Set<CompletionItem> {
+        switch self {
+        case let .compound(items): return items
+        default: return Set([self])
+        }
+    }
+}
+
+struct Completion {
+    let item: CompletionItem
+    let standardPipelines: StandardPipelines
+    let terminal: Terminaling
+    let theme: Theme
+
+    func run() {
+        let flattened = item.flattened
+        for (index, item) in flattened.enumerated() {
+            let textAndNexts: [(text: TerminalText, next: TerminalText?)] = switch item.message {
+            case let .list(messages): messages.map(\.textAndNext!)
+            case .string: [item.message.textAndNext!]
+            }
+
+            if textAndNexts.isEmpty { continue }
+
+            let standardPipeline = switch item {
+            case .error: standardPipelines.error
+            default: standardPipelines.output
+            }
+
+            let (title, color) = switch item {
+            case .error: ("[ Error ]", theme.danger)
+            case .warning: ("[ Warning ]", theme.accent)
+            case .success: ("[ Success ]", theme.success)
+            case .compound: fatalError("Unexpected nested compound message inside another compound message.")
+            }
+
+            standardPipeline.write(content: "\(title)\n".boldIfColoredTerminal(terminal).hexIfColoredTerminal(color, terminal))
+
+            if textAndNexts.count == 1 {
+                let (text, next) = textAndNexts.first!
+                standardPipeline.write(content: "\(text.description) \n".hexIfColoredTerminal(color, terminal))
+                if let next {
+                    standardPipeline.write(content: "\n  Suggestion: \n".hexIfColoredTerminal(color, terminal))
+                    for nextLine in next.description.split(separator: "\n") {
+                        standardPipeline.write(content: "    ▸ \(nextLine)\n".hexIfColoredTerminal(color, terminal))
+                    }
+                }
+            } else {
+                for (index, (text, next)) in textAndNexts.enumerated() {
+                    standardPipeline
+                        .write(content: "  ▸ \(index + 1). \(text.description) \n".hexIfColoredTerminal(color, terminal))
+                    if let next {
+                        for (nextLineIndex, nextLine) in next.description.split(separator: "\n").enumerated() {
+                            if nextLineIndex == 0 {
+                                standardPipeline.write(content: "     ↳ \(nextLine) \n".hexIfColoredTerminal(color, terminal))
+                            } else {
+                                standardPipeline.write(content: "       \(nextLine) \n".hexIfColoredTerminal(color, terminal))
+                            }
+                        }
+                    }
+                }
```

Comment:
I wonder if we should force things to be flat. What's an example of a message where it makes sense to go deeper than one level? I also don't see any tests for this output, but maybe I've missed it.